main.tf

```diff
@@ -26,18 +26,23 @@ resource "azurerm_kubernetes_cluster" "main" {
     name            = "nodepool"
     node_count      = var.agents_count
     vm_size         = var.agents_size
-    os_disk_size_gb = 50
+    os_disk_size_gb = var.os_disk_size_gb
+    vnet_subnet_id  = var.vnet_subnet_id
```

Comment:
if users don't specify "vnet_subnet_id" when using this module, it defaults to null, will there be some errors ?

Reply:
Not yet. null is accepted for `vnet_subnet_id`, I have just tested it.